tests/fetchai/test_communication.py

```diff
@@ -0,0 +1,74 @@
+import json
+import uuid
+
+import pytest
+from requests_mock import Mocker as RequestsMocker
+
+from fetchai import communication
+from fetchai.crypto import Identity
+
+
+@pytest.fixture
+def identity() -> Identity:
+    return Identity.from_seed("TESTING", 1)
+
+
+def fake_agent_lookup_function(agent_address: str) -> str:
+    """A fake agent look up function to use in test.
+
+    Returns a predictable URL for testing against"""
+
+    return f"http://localhost/fake_endpoint/{agent_address}"
+
+
+class TestSendMessageToAgent:
+    def test_happy_path(self, identity: Identity):
+        agent_name = "agent_one"
+        expected_url = "http://localhost/fake_endpoint/agent_one"
+
+        with RequestsMocker() as mock:
+            mock.post(expected_url)
+            communication.send_message_to_agent(
+                identity,
+                agent_name,
+                {},
+                agent_lookup_function=fake_agent_lookup_function,
+            )
+
+            assert mock.called_once
```

Comment:
```suggestion
        assert mock.called_once
        assert mock.last_request.scheme == "http"
        assert mock.last_request.netloc == "localhost"
        assert mock.last_request.port == 80
        assert mock.last_request.path == f"/fake_endpoint/{agent_name}"

        payload = communication.Envelope.model_validate_json(mock.last_request.text)
        assert (
            payload.protocol_digest
            == "proto:a03398ea81d7aaaf67e72940937676eae0d019f8e1d8b5efbadfef9fd2e98bb2"
        )
        assert (
            payload.sender
            == "agent1qv8qnqv4ddslkktlalqhv4zz7wsjnwslv0h33gpwh7fhw3txae2lg89awsp"
        )
        assert payload.target == "agent_one"
        assert payload.payload == "e30="
```
Just some optional add-ins that ensure we're sending the correct payload as well

Reply:
If you want to test these - we should probably split them out to a `test_whatever_is_called_correctly`